File: patch_livepanel.py
import re

with open('src/components/LivePanel.tsx', 'r') as f:
    content = f.read()

# Replace the plain text div with a styled one
old_code = """                        {viewMode !== 'summary' && (
                          <div className={`${viewMode === 'large' ? 'p-3 text-sm min-h-[64px]' : viewMode === 'small' ? 'p-1.5 text-[10px] min-h-[32px]' : 'p-2 text-[11px] min-h-[48px]'} text-gray-200 leading-relaxed font-sans whitespace-pre-line bg-[#1c1e24]`}>
                            {slide.text}
                          </div>
                        )}"""

new_code = """                        {viewMode !== 'summary' && (
                          <div 
                            className={`${viewMode === 'large' ? 'p-3 text-sm min-h-[64px]' : viewMode === 'small' ? 'p-1.5 text-[10px] min-h-[32px]' : 'p-2 text-[11px] min-h-[48px]'} leading-relaxed font-sans whitespace-pre-line bg-[#1c1e24] bg-cover bg-center relative overflow-hidden`}
                            style={{
                              backgroundImage: slide.backgroundUrl ? (slide.backgroundUrl.startsWith('linear-gradient') ? slide.backgroundUrl : `url(${slide.backgroundUrl})`) : undefined,
                            }}
                          >
                            {slide.backgroundUrl && <div className="absolute inset-0 bg-black/40 pointer-events-none" />}
                            <div 
                              className="relative z-10"
                              style={{
                                textTransform: (isSong && (systemOptions?.mainOutput?.song?.allCapsLyrics || systemOptions?.mainOutput?.song?.songFont?.casing === 'uppercase')) ? 'uppercase' : undefined,
                                color: slide.backgroundUrl ? '#ffffff' : '#e5e7eb',
                                textShadow: slide.backgroundUrl ? '0px 1px 4px rgba(0,0,0,0.9)' : undefined,
                                textAlign: 'center',
                                fontWeight: '700'
                              }}
                            >
                              {slide.text}
                            </div>
                          </div>
                        )}"""

if old_code in content:
    content = content.replace(old_code, new_code)
    with open('src/components/LivePanel.tsx', 'w') as f:
        f.write(content)
    print("Patched LivePanel.tsx successfully!")
else:
    print("Could not find old_code in LivePanel.tsx")

File: patch_live_panel.js
const fs = require('fs');
let code = fs.readFileSync('src/components/LivePanel.tsx', 'utf8');

const targetStr = `{isScripture && slide.verses && slide.verses.length > 0 ? (
                                slide.verses.map((v: any, vIdx: number) => (
                                  <span key={v.verse || vIdx} className="inline">
                                    {(systemOptions?.mainOutput?.scripture?.showVerseNumbers ?? true) && (
                                      <span 
                                        className="font-bold inline-block mr-1.5 select-none"
                                        style={{ 
                                          color: systemOptions?.mainOutput?.scripture?.verseFont?.color || systemOptions?.mainOutput?.scripture?.verseColor || '#F6E05E',
                                          fontFamily: systemOptions?.mainOutput?.scripture?.verseFont?.family || resolvedStyles.fontFamily || 'Tahoma, sans-serif',
                                        }}
                                      >
                                        {formatVerseNumber(v.verse, systemOptions?.mainOutput?.scripture?.verseNumberStyle)}
                                      </span>
                                    )}
                                    <span>{v.text}</span>
                                    {vIdx < slide.verses.length - 1 && ' '}
                                  </span>
                                ))
                              ) : (
                                slide.text
                              )}`;

const replacementStr = `{liveItem?.type === 'camera' ? (
                                <div className="flex flex-col items-center justify-center opacity-80 h-full py-2">
                                  <MonitorUp size={24} className="mb-1 text-emerald-400" />
                                  <span className="text-[10px] font-mono text-emerald-300 font-semibold tracking-wider uppercase">{liveItem.name} • LIVE</span>
                                </div>
                              ) : liveItem?.type === 'video' ? (
                                <div className="flex flex-col items-center justify-center opacity-80 h-full py-2">
                                  <Film size={24} className="mb-1 text-cyan-400" />
                                  <span className="text-[10px] font-mono text-cyan-300 font-semibold tracking-wider uppercase truncate max-w-full px-2">
                                    {liveItem.name} • {liveItem.data?.url?.split('.').pop()?.toUpperCase() || 'VIDEO'}
                                  </span>
                                </div>
                              ) : liveItem?.type === 'audio' ? (
                                <div className="flex flex-col items-center justify-center opacity-80 h-full py-2">
                                  <Volume2 size={24} className="mb-1 text-purple-400" />
                                  <span className="text-[10px] font-mono text-purple-300 font-semibold tracking-wider uppercase truncate max-w-full px-2">
                                    {liveItem.name} • {liveItem.data?.url?.split('.').pop()?.toUpperCase() || 'AUDIO'}
                                  </span>
                                </div>
                              ) : liveItem?.type === 'image' ? (
                                <div className="absolute inset-0 flex items-center justify-center">
                                  {slide.backgroundUrl ? (
                                    <img src={slide.backgroundUrl} alt="Preview" className="w-full h-full object-contain bg-black/60" />
                                  ) : (
                                    <span className="text-[10px] font-mono opacity-50">{liveItem.name}</span>
                                  )}
                                </div>
                              ) : liveItem?.type === 'presentation' ? (
                                <div className="absolute inset-0 bg-black overflow-hidden">
                                  {liveItem.data?.fileBytes ? (
                                    <React.Suspense fallback={<div className="flex h-full items-center justify-center text-[10px] text-white/30 font-mono">Loading...</div>}>
                                      <div className="w-full h-full pointer-events-none">
                                        <PptxRenderOverlay fileBytes={liveItem.data.fileBytes} activeSlideIndex={idx} />
                                      </div>
                                    </React.Suspense>
                                  ) : (
                                    <div className="flex h-full items-center justify-center text-[10px] text-white/50 p-2 text-center">
                                      {slide.title || 'Slide'}
                                    </div>
                                  )}
                                </div>
                              ) : isScripture && slide.verses && slide.verses.length > 0 ? (
                                slide.verses.map((v: any, vIdx: number) => (
                                  <span key={v.verse || vIdx} className="inline">
                                    {(systemOptions?.mainOutput?.scripture?.showVerseNumbers ?? true) && (
                                      <span 
                                        className="font-bold inline-block mr-1.5 select-none"
                                        style={{ 
                                          color: systemOptions?.mainOutput?.scripture?.verseFont?.color || systemOptions?.mainOutput?.scripture?.verseColor || '#F6E05E',
                                          fontFamily: systemOptions?.mainOutput?.scripture?.verseFont?.family || resolvedStyles.fontFamily || 'Tahoma, sans-serif',
                                        }}
                                      >
                                        {formatVerseNumber(v.verse, systemOptions?.mainOutput?.scripture?.verseNumberStyle)}
                                      </span>
                                    )}
                                    <span>{v.text}</span>
                                    {vIdx < slide.verses.length - 1 && ' '}
                                  </span>
                                ))
                              ) : (
                                slide.text
                              )}`;

if (code.indexOf(targetStr) === -1) {
  console.log("TARGET STRING NOT FOUND!");
} else {
  // Replace all occurrences (there should be 2)
  let newCode = code.split(targetStr).join(replacementStr);
  fs.writeFileSync('src/components/LivePanel.tsx', newCode);
  console.log("REPLACED SUCCESSFULLY!");
}
